packages/pharos/src/components/coach-mark/pharos-coach-mark.ts

```diff
@@ -0,0 +1,166 @@
+import { PharosElement } from '../base/pharos-element';
+import { PharosButton } from '../button/pharos-button';
+import { PharosHeading } from '../heading/pharos-heading';
+import { html } from 'lit';
+import { property } from 'lit/decorators.js';
+import type { TemplateResult, CSSResultArray } from 'lit';
+import { coachMarkStyles } from './pharos-coach-mark.css';
+import ScopedRegistryMixin from '../../utils/mixins/scoped-registry';
+import debounce from '../../utils/debounce';
+
+export type Side = 'top' | 'right' | 'bottom' | 'left';
+export type Alignment = 'start' | 'center' | 'end';
+export type Delay = 'none' | 'short' | 'long';
+export type Variant = 'light' | 'dark';
+
+/**
+ * Pharos coach mark component.
+ *
+ * @tag pharos-coach-mark
+ *
+ */
+export class PharosCoachMark extends ScopedRegistryMixin(PharosElement) {
+  static elementDefinitions = {
+    'pharos-button': PharosButton,
+    'pharos-heading': PharosHeading,
+  };
+
+  private offsetX = 0;
+  private offsetY = 0;
+  private margin = 20;
+
+  public static override get styles(): CSSResultArray {
+    return [coachMarkStyles];
+  }
+
+  /**
+   * Indicates that the coach mark should not be displayed on the page
+   * @attr hide
+   */
+  @property({ type: Boolean, reflect: true })
+  public hide = true;
+
+  /**
+   * Indicates which side of the base element the coachmark should appear on
+   * @attr side
+   * @type {Side}
+   */
+  @property({ reflect: true })
+  public side: Side = 'bottom';
+
+  /**
+   * Indicates which side of the base element the coachmark should appear on
+   * @attr alignment
+   * @type {Alignment}
+   */
+  @property({ reflect: true })
+  public alignment: Alignment = 'center';
+
+  /**
+   * Dictates how long to wait between coach mark trigger and start of coach mark fade in animation
+   * @attr delay
+   * @type {Delay}
+   */
+  @property({ reflect: true })
+  public delay: Delay = 'short';
+
+  /**
+   * Text content for the modal header
+   * @attr header
+   * @type {String}
+   */
+  @property({ reflect: true })
+  public header = '';
+
+  /**
+   * Style variant
+   * @attr variant
+   * @type {Variant}
+   */
+  @property({ reflect: true })
+  public variant: Variant = 'light';
+
+  /**
+   * Set minimum width of coach mark
+   * @attr width
+   * @type {number}
+   */
+  @property({ reflect: true })
+  public width = 250;
+
+  constructor() {
+    super();
+    this.setOffset();
+    this.resizeObserver.observe(document.body);
+  }
+
+  private resizeObserver = new ResizeObserver(
+    debounce((entries) => {
+      entries.map((entry: ResizeObserverEntry) => {
+        if (entry.target.tagName.toLowerCase() !== 'body') return;
```

Comment:
is this check necessary since only the body tag is being observed?

Reply:
It is not required, removed in [9af8d05](https://github.com/ithaka/pharos/pull/608/commits/9af8d05bfa5f9cc3d76e79874fd310c1fa7c1394)